src/megatron/bridge/models/model_provider.py

```diff
@@ -551,8 +560,8 @@ def get_model(
         for model_module in model:
             model_module.cuda(torch.cuda.current_device())
 
-    if model_config.fp16 or model_config.bf16:
-        model = [Float16Module(model_config, model_module) for model_module in model]
+    if (model_config.fp16 or model_config.bf16) and mixed_precision_wrapper is not None:
```

Comment:
is there a risk that fp16/bf16 is enabled and mixed_precision_wrapper is not specified causing this to silently fall through? should this assert instead that if the mixed precision flags are specified that the wrapper module must be set?